mediabridge/api/app.py

```diff
@@ -51,6 +52,23 @@ def search_movies() -> tuple[Response, int]:
             movies_list = [row._asdict() for row in movies]
             return jsonify(movies_list), 200
 
+    @app.route("/api/v1/movie/recommend", methods=["POST"])  # type: ignore
+    def recommend_movies() -> tuple[Response, int]:
+        data = request.get_json()
+        if not data or "movies" not in data:
+            return jsonify({"error": "JSON body with 'movies' array is required."}), 400
+        movies = data["movies"]
+        if not isinstance(movies, list) or not all(isinstance(m, str) for m in movies):
+            return jsonify({"error": "'movies' must be a list of strings."}), 400
+
+        try:
+            rec_ids = recommend()
+            rec_titles = [get_title(int(mid)) for mid in rec_ids]
+        except Exception as e:
```

Comment:
It's considered bad practice to `except Exception`, because if there are logical errors in your code (`NameError`, `IndexError`, etc), they will also get caught here. Prefer to catch the precise exception you're worried about.